### PR TITLE
Naming conventions for quickstarters

### PR DESCRIPTION
**Pull request by adyachok**
Introduced logic for  building quickstarter naming convention .

1. Created configuration file with requirements for names selection accordingly to project names supported by different frameworks.
2. Created endpoint for front-end to extract information about specified naming rules.
3. Created filtering logic for processes of creation of a new project as well as update existing one.

### CODE REVIEW

**Comment by gerardcl** [comment]
LGTM! I will test it and let's have it into 1.2.x too!

**Comment by adyachok** [comment]
Sure. I will add tests.
I think we need to fix FE part as well. I can include regex for FE part in a rules object, however I would like to invite a front-end developer for an increment in FE part.

**Comment by clemensutschig** [comment]
1.2 i guess.. ;-)

**Comment by gerardcl** [comment]
hi @adyachok and all, is this ok if we do not put this feature yet in ODS1.2.x initial release but later on? I want to assure we are not adding any special change at this moment, does it make sense?

**Comment by stefanlack** [comment]
Created https://github.com/adyachok/ods-provisioning-app/pull/1

**Comment by adyachok** [comment]
@stefanlack I nice work.

**Comment by clemensutschig** [comment]
@stefanlack - go for 2.0?

**Comment by stefanlack** [comment]
@clemensutschig I'm afraid but I do not understand your last comment. I have created an PR on Oct.7, @adyachok answered with "nice work". So @adyachok  and you may merge my pr into #234  and finish this.

**Comment by michaelsauter** [comment]
I propose to move this to v3.

**Comment by stefanlack** [comment]
I also propose to move this to v3

File: src/main/java/org/opendevstack/provision/controller/ProjectApiController.java [hunk 1]
```diff
@@ -209,6 +217,58 @@
...
   }
 
+  private List<Map<String, String>> filterQuickstarters(List<Map<String, String>> quickstarters) {
+    /**
+     * Pushes every quckstarter object though chain of filters.
+     *
+     * @param quickstarters - selected by user quickstarters to create
+     * @return list of correctly named parameters
+     */
+    List<Job> availableQuickstarters = rundeckAdapter.getQuickstarters();
```
Comment (stitakis) [inline src/main/java/org/opendevstack/provision/controller/ProjectApiController.java:227]: rundeck is deprecated and not supported anymore

File: src/test/java/org/opendevstack/provision/controller/ProjectApiControllerTest.java [hunk 2]
```diff
@@ -108,9 +125,14 @@ private void initOpenProjectData() {
     data.description = "Description";
 
     Map<String, String> someQuickstarter = new HashMap<>();
-    someQuickstarter.put("key", "value");
+    someQuickstarter.put("component_id", "fe-angular-7");
```
Comment (stitakis) [inline src/test/java/org/opendevstack/provision/controller/ProjectApiControllerTest.java:128]: numbers are not supported anymore in component id

**Comment by stitakis** [comment]
I haven't taken a deep look into this PR, but first impression is that needs to be "adapter" to latest version from master.

**Comment by michaelsauter** [comment]
Let's close this. The problem this PR addresses is fixed by other means - without the extra complexity. Let's come back to it when we have users needing this.

**Pull request closed without merging**